bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/PasteAction.java

```diff
@@ -99,6 +107,22 @@ private IResource getTarget() {
 		return null;
 	}
 
+	/**
+	 * @param selectedResources currently selected resources
+	 * @param clipboardContent  current content of the clipboard
+	 * @return true if selected resources are the same as clipboard content
+	 */
+	private boolean selectionIsSameAsClipboardContent(List<? extends IResource> selectedResources,
+			List<? extends IResource> clipboardContent) {
+		if (clipboardContent != null && selectedResources != null && !selectedResources.isEmpty()) {
+			if (selectedResources.size() != clipboardContent.size()) {
+				return false;
+			}
+			return selectedResources.containsAll(clipboardContent);
+		}
+		return false;
+	}
```

Comment:
This method is an "equals" but it allows for different order of the items, so I think this more generic version is more appropriate:

```suggestion
	/**
	 * @param a The first collection
	 * @param b The second collection
	 * @return true if both collections aren't null and they contain the exact same
	 *         items, regardless of their order.
	 */
	private boolean areEqualsUnordered(Collection<? extends IResource> a, Collection<? extends IResource> b) {
		return b != null && a != null && !a.isEmpty() // they are not empty...
				&& a.size() == b.size() // ... and they have the same size
				&& a.containsAll(b); // ... and all elements of A in B
	}
```

If performance ever becomes an issue (please test for it) then you could change the implementation so it first sorts both lists (or a copy of the list if you don't want to change the original lists) and then calls `equals` on them.

Reply:
Thanks for the feedback, I'll change that

Reply:
@alexanderKononov-sap: Did you have a look on performance. Was the proposal "first sort and then call equals" not necesarry?

Reply:
@BeckerWdf performance is still good, I checked by copying a folder with 100 classes into itself